### PR TITLE
Add better error to objax.GradValues function for input_argnums

### PR DESCRIPTION
**Pull request by carlini**
Currently `input_argnums` expects a tuple. If you pass the integer 0, then the `or` treats this as if you passed nothing, and so  it passes an empty tuple and silently does the wrong thing. This fixes that to check the type.

### CODE REVIEW

File: objax/gradient.py [hunk 1]
```diff
@@ -60,7 +60,13 @@ def f_func(inputs_and_train_tensors: List[JaxArray],
             finally:
                 self.vc.assign(original_vc)
 
-        self.input_argnums = input_argnums or tuple()
+        if input_argnums is None:
+            self.input_argnums = tuple()
+        else:
+            assert isinstance(input_argnums, tuple), \
+                f"Must pass a tuple of indices to input_argnums; received {input_argnums}."
+            self.input_argnums = input_argnums
```
Comment (david-berthelot) [inline objax/gradient.py:68]: I don't think we need an if statement, we can keep the code simple by changing the assert condition.
```suggestion
        assert isinstance(input_argnums, tuple) or input_argnums is None, \
                f'Must pass a tuple of indices to input_argnums; received {input_argnums}.'
        self.input_argnums = input_argnums or tuple()
```

Also nit, let's use single quotes to keep string consistency with rest of code.

**Review by david-berthelot — CHANGES_REQUESTED** [review]

File: objax/gradient.py [hunk 2]
```diff
@@ -60,7 +60,10 @@ def f_func(inputs_and_train_tensors: List[JaxArray],
             finally:
                 self.vc.assign(original_vc)
 
+        assert isinstance(input_argnums, tuple) or input_argnums is None, \
+                f'Must pass a tuple of indices to input_argnums; received {input_argnums}.'
```
Comment (david-berthelot) [inline objax/gradient.py:64]: There's a flake8 error. Can you fix it?

**Review by david-berthelot — CHANGES_REQUESTED** [review]

**Comment by carlini** [comment]
Pushing fix

File: objax/io/checkpoint.py [hunk 3]
```diff
@@ -89,6 +89,8 @@ def restore(self, vc: VarCollection, idx: Optional[int] = None):
             idx: index of the restored checkpoint.
             ckpt: full path to the restored checkpoint.
         """
+        assert isinstance(vc, VarCollection), \
+            f"Must pass a VarCollection to save; received type {type(vc)}."
```
Comment (david-berthelot) [inline objax/io/checkpoint.py:93]: Nit: coding style consistency, also I believe you meant restore not save here.
```suggestion
            f'Must pass a VarCollection to restore; received type {type(vc)}.'
```

**Review by david-berthelot — CHANGES_REQUESTED** [review]

File: objax/io/checkpoint.py [hunk 4]
```diff
@@ -109,6 +111,8 @@ def save(self, vc: VarCollection, idx: int):
             vc: variables collection to save.
             idx: index of the new checkpoint where variables should be saved.
         """
+        assert isinstance(vc, VarCollection), \
+            f"Must pass a VarCollection to save; received type {type(vc)}."
```
Comment (david-berthelot) [inline objax/io/checkpoint.py:115]:
```suggestion
            f'Must pass a VarCollection to save; received type {type(vc)}.'
```

**Review by david-berthelot — CHANGES_REQUESTED** [review]

**Review by david-berthelot — APPROVED** [review]
